modules/post/windows/gather/checkvm.rb

```diff
@@ -36,16 +41,25 @@ module supports detection of Hyper-V, VMWare, Virtual PC,
   end
 
   def get_services
-    @services ||= registry_enumkeys('HKLM\SYSTEM\ControlSet001\Services')
+    @services ||= registry_enumkeys('HKLM\\SYSTEM\\ControlSet001\\Services')
     @services
   end
 
   def service_exists?(service)
     get_services && get_services.include?(service)
   end
 
+  def get_regval_str(key, valname)
+    ret = registry_getvaldata(key, valname)
+    if ret.kind_of(Array)
```

Comment:
Parallels detection works after applying this change 👍 
```suggestion
    if ret.kind_of?(Array)
```
```
msf6 post(windows/gather/checkvm) > rexploit
[*] Reloading module...

[*] Checking if the target is a Virtual Machine ...
[+] This is a Parallels Virtual Machine
[*] Post module execution completed
```